Updates Django syntax highlighting, fixes bugs, and adding django 4 commands

scuml opened the pull request:
Updates syntax highlighter for HTML (Django) to latest format, updating the template tags and filters, and inheriting from the sublime HTML package.  This fixes quite a few regex bugs.

scuml commented:
Great - thank you for letting me know.  This will fix things like syntax highlighting not reverting back to html after `<script>` tags, and also html-tags with dashes like `<turbo-frame>`.  

Enjoy your vacation.

squ1b3r commented on Syntaxes/HTML (Django).sublime-syntax:
```diff
@@ -0,0 +1,241 @@
+        thumbnail|endthumbnail|\
+        static|get_static_prefix|get_media_prefix)"
+
+  template_filters: "(
```
This is missing filters from https://docs.djangoproject.com/en/4.1/ref/contrib/humanize/

squ1b3r commented on Syntaxes/HTML (Django).tmLanguage:
```diff
@@ -1,1204 +0,0 @@
-<?xml version="1.0" encoding="UTF-8"?>
```
Along with this, you can also remove `HTML (Django).JSON-tmLanguage`. It was the source file that this file was generated from.

squ1b3r commented on Syntaxes/HTML (Django).sublime-syntax:
```diff
@@ -0,0 +1,241 @@
+    - match: "(?=[^\\w])"
+      pop: true
+
+
```
nit: extra line

squ1b3r commented on Syntaxes/HTML (Django).sublime-syntax:
```diff
@@ -0,0 +1,241 @@
+
+  tagname:
+
+
```
nit: extra line

squ1b3r commented on Syntaxes/HTML (Django).sublime-syntax:
```diff
@@ -0,0 +1,241 @@
+            - include: tagname
+
+
+
```
nit: extra line and unnecessary comment below it.

squ1b3r commented on Syntaxes/HTML (Django).sublime-syntax:
```diff
@@ -0,0 +1,241 @@
+            - meta_scope: text.django comment.block.django
+            - include: commentblock
+
+
```
nit: extra line

balazs-endresz commented on README.md:
```diff
@@ -1,4 +1,4 @@
-# Django support for Sublime Text 2/3
+# Django support for Sublime Text 4
```
`.sublime-syntax` was added in ST3: https://docs.sublimetext.io/guide/extensibility/syntaxdefs.html  I've also tried this and it does work with version 3.

balazs-endresz commented on Syntaxes/HTML (Django).sublime-syntax:
```diff
@@ -0,0 +1,241 @@
+first_line_match: (?i){%\s?(?>load|extends)\s
+
+variables:
+  template_tags: "(
```
Would be possible to match whole words only? This doesn't work well in some cases: https://i.imgur.com/5XzkGOo.png

balazs-endresz replied:
If I add `- match: '\b{{template_tags}}\b'` (with single quotes) that does the whole word matching for me but only if I also make the value of `template_tags` a single line string. Otherwise only words that are the last one on each line (in the definition) are highlighted, e.g. `endfor` matches but not `for`. This is with ST3.

balazs-endresz replied:
Actually, `- match: '\b{{template_tags}}\b'` does work just fine! 

(The problem for me was that I also made the definition of `template_tags` use single quotes, but that has to remain with double quotes. I don't know how this works exactly.)

balazs-endresz replied:
Thank you for putting this together! However, the [current version](https://github.com/squ1b3r/Djaneiro/blob/4e91987b689702260afc2f471a32441d2f4d08c6/Syntaxes/HTML%20(Django).sublime-syntax) now doesn't work at all: https://i.imgur.com/ruCH0sT.png I still only tried on ST3.

scuml replied:
Looks like this is due to using a pop command that supports only build >4050

scuml commented:
Thanks for the great code reviews.  Fixed the issues noted and updated a sample django template file @squ1b3r 
[sample_django.txt](https://github.com/squ1b3r/Djaneiro/files/9447096/sample_django.txt)


Adding a screenshot to save time.
![Screen Shot 2022-08-29 at 2 41 57 PM](https://user-images.githubusercontent.com/1790447/187274614-8c58a7c3-6ae4-4f51-acf7-eacc20cf54c3.png)

scuml commented:
@squ1b3r The latest fixes to the parser uses a Sublime 4-only command to pop scopes.  My recommendation is a link in the README to manually download the ST3 version, which will allow future updates to make full use of the Sublime 4 API improvements.  Could you let me know your thoughts?

squ1b3r commented:
> @squ1b3r The latest fixes to the parser uses a Sublime 4-only command to pop scopes. My recommendation is a link in the README to manually download the ST3 version, which will allow future updates to make full use of the Sublime 4 API improvements. Could you let me know your thoughts?

Hi there, sorry for being absent again. Got back from vacation and got covid. Fun times. 

Just checked and the first version of Sublime Text 4 was released in May 2021. That's more than a year ago. At this point, I would assume the majority already uses version 4 but to make sure things still work for people who use versions <4 we can rename the current syntax file to something else and keep it in the repo for now maybe? The new one will be activated by default, but people will still have the option to use the old one if they need to.

scuml commented:
That's a good idea @squ1b3r .  Added the previous syntax file and made a note in the README.

squ1b3r commented:
@scuml Thank you so much for your contribution! Hopefully, this is not going to brick anything and everything goes well.
